src/controllers/user-settings/edit-exclude-categories.ts

```diff
@@ -0,0 +1,33 @@
+import { API_RESPONSE_STATUS } from 'shared-types';
+import { z } from 'zod';
+import { editExcludedCategories } from '@services/user-settings/edit-excluded-categories';
+import { errorHandler } from '@controllers/helpers';
+import { CustomResponse } from '@common/types';
+
+export const editExcludedCategoriesHandler = async (
+  req,
+  res: CustomResponse
+) => {
+  try {
+    const { addIds, removeIds } = editExcludedCategoriesSchema.parse(req.body);
+    const { user } = req;
```

Comment:
You should't parse it manually, there's a `req.validated` field if you use middleware